src/Container.php

```diff
@@ -16,6 +16,25 @@
 use WCPay\Internal\DependencyManagement\ServiceProvider\GenericServiceProvider;
 use WCPay\Internal\DependencyManagement\ServiceProvider\ProxiesServiceProvider;
 
+/**
+ * Hides errors during update from 6.6.0 or 6.6.1 to 6.6.2.
+ *
+ * This class would be loaded without the right dependencies (and autoloader)
+ * being loaded before it after the update is complete. When that happens,
+ * the ContainerInterface would still be in a different namespace, and would not exist here.
+ *
+ * Preventing the class from being loaded here does nothing but hide the error.
+ * All later requests will work properly.
+ */
+if (
+	! interface_exists( ContainerInterface::class )
+	&& isset( $_GET['action'] ) // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	&& 'upload-plugin' === $_GET['action'] // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	&& 'update.php' === $GLOBALS['pagenow']
```

Comment:
May check `isset` for `pagenow` too?

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/7525/commits/c6d4d49d870f15750036c29d32a8ac64db561890